### PR TITLE
Update various dependencies

### PR DESCRIPTION
**Pull request by aljones15**
1. Update peerDependencies for `bedrock-security-context` to `^4.0.0`.
2. Update `engines.node` to `>= 14.0.0`
3. Update devDeps for `eslint` and `eslint-config-digitalbazaar` to latest.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,13 +27,13 @@
   "peerDependencies": {
     "bedrock": "2.0.0 - 4.x",
     "bedrock-jsonld-document-loader": "^1.0.0",
-    "bedrock-security-context": "^3.0.0"
+    "bedrock-security-context": "^4.1.0"
   },
   "engines": {
-    "node": ">=6.3.0"
+    "node": ">=14.0.0"
```
Comment (dlongley) [inline package.json]: Why this change and not node 12?

Comment (aljones15) [reply package.json]: https://github.com/digitalbazaar/bedrock-ledger-context/pull/14/commits/3b1d5b5c4aee9781f537352b936e27a7c1abcf16

File: package.json [hunk 2]
```diff
@@ -27,13 +27,13 @@
   "peerDependencies": {
     "bedrock": "2.0.0 - 4.x",
```
Comment (dlongley) [inline package.json]: This should be just 4.x, we want to avoid ranges.

**Review by dlongley — APPROVED** [review]
Thanks! LGTM, other than the comment I made.

File: package.json (hunk 2 above)
Comment (aljones15) [reply package.json]: Addressed here: https://github.com/digitalbazaar/bedrock-ledger-context/pull/14/commits/3ee49a21f51bab4860dc82b8ef94c90133b4623a
